runtime/src/accounts_db.rs

```diff
@@ -4339,9 +4347,9 @@ impl AccountsDb {
     pub fn calculate_accounts_hash_without_index(
         storages: &[SnapshotStorage],
         thread_pool: Option<&ThreadPool>,
+        mut stats: HashStats,
```

Comment:
Where is this used?

Reply:
this function passes it to AccountsHash::rest_of_hash_calculation.
rest_of_hash_calculation eventually logs the datapoint after the last pass:
```
calculate_accounts_hash_without_index accounts_scan=3310614i eliminate_zeros=403638i hash=242073i sort=0i hash_total=61837974i flatten=0i storage_sort_us=14883i unreduced_entries=0i num_snapshot_storage=372887i total=3971208i
```

Reply:
I see, you replace the stats which was defined locally -- I mistook it was totally new. Any reason why it is not passed in by reference &mut HashStats which I thought is more efficient?

Reply:
I want to transfer ownership in this case, so the function takes the stats by value. I imagine this is passing a reference in generated code anyway. You could verify this with the rust code generator webpage. Also, this is called once per many minutes. Another alternative was for me to pass just this u64 time into the function for it to add to its stats. I have found passing the stats struct around is the preferred method. It allows for measurements to be added more easily and reduces # of parameters to functions.
Also, struct HashStats does not implement Copy, so no copy can be made or there would be a compile error.
Feel free to ask followups on any of these assertions I've made. ;-)

Reply:
sounds good. So the compiler can do optimization on this for you?

Reply:
lijun and I resolved this offline and with godbolt.